web_poet/pages.py

```diff
@@ -64,9 +71,80 @@ def __init_subclass__(cls, skip_nonitem_fields: bool = False, **kwargs):
     async def to_item(self) -> ItemT:
         """Extract an item from a web page"""
         return await item_from_fields(
-            self, item_cls=self.item_cls, skip_nonitem_fields=self._skip_nonitem_fields
+            self,
+            item_cls=self.item_cls,
+            skip_nonitem_fields=self._skip_nonitem_fields,
+            field_names=self.fields_to_extract,
         )
 
+    def _get_select_fields(self) -> Optional[SelectFields]:
+        select_fields = getattr(self, "select_fields", None)
+        if not select_fields:
+            return None
+
+        if not isinstance(select_fields, SelectFields):
+            return None
+
+        return select_fields
+
+    @property
+    def fields_to_extract(self) -> Iterable[str]:
+        """Returns an Iterable of field names which should populate the designated
+        ``item_cls``.
+
+        If :class:`web_poet.fields.SelectFields` is set, this takes into account
+        the fields that are marked as included or excluded alongside any field
+        that are marked as enabled/disabled by default.
+        """
+        select_fields = self._get_select_fields()
+        if select_fields is None:
+            return list(get_fields_dict(self))
+
+        fields = select_fields.fields
+
+        if fields is None or len(fields) == 0:
+            return list(get_fields_dict(self))
+
+        page_obj_fields = get_fields_dict(self, include_disabled=True)
+
+        unknown_fields = set(fields) - set(page_obj_fields.keys()).union({"*"})
+        if unknown_fields:
+            self._handle_unknown_field(unknown_fields, select_fields.on_unknown_field)
```

Comment:
The handling of unknown fields seems to be more optimal to be placed here as opposed to something like `item_from_fields()` which could've already extracted some fields (which could be expensive, i.e., having additional request) before erroring out.